internal/graphql/model/helpers.go

```diff
@@ -114,3 +123,121 @@ func applyYaml(client *mesherykube.Client, delete bool, file string) error {
 
 	return nil
 }
+
+// installs operator
+func installUsingHelm(client *mesherykube.Client, delete bool, adapterTracker models.AdaptersTrackerInterface) error {
+	// retrieving meshery's version to apply the appropriate chart
+	mesheryReleaseVersion := viper.GetString("BUILD")
+	if mesheryReleaseVersion == "" || mesheryReleaseVersion == "Not Set" || mesheryReleaseVersion == "edge-latest" {
+		latestReleaseData, err := handlers.CheckLatestVersion("")
+		if err != nil {
+			mesheryReleaseVersion = latestReleaseData.Current
+		}
+	}
+
+	var (
+		act   = mesherykube.INSTALL
+		chart = "meshery-operator"
+	)
+
+	// a basic check to see if meshery is installed in cluster
+	// this helps decide what chart should be used for installing operator
+	if viper.GetString("KUBERNETES_SERVICE_HOST") != "" {
+		act = mesherykube.UPGRADE
+		chart = "meshery"
+	} else if delete {
+		act = mesherykube.UNINSTALL
+	}
+
+	overrides, err := SetOverrideValues(delete, adapterTracker)
+	if err != nil {
+		return err
+	}
+
+	err = client.ApplyHelmChart(mesherykube.ApplyHelmChartConfig{
+		Namespace: "meshery",
+		ChartLocation: mesherykube.HelmChartLocation{
+			Repository: chartRepo,
+			Chart:      chart,
+			Version:    mesheryReleaseVersion,
+		},
+		CreateNamespace: true,
```

Comment:
@DelusionalOptimist, should this flag be set to true, even when the action is UNINSTALL ?

Reply:
@hexxdump this flag is actually meaningless when the action is UNINSTALL. It doesn't have any effect on the ns. But it sure is confusing... I'll add a comment :+1:

Reply:
thanks for clarifying that @DelusionalOptimist !